syscalls_linux.go

```diff
@@ -22,7 +22,7 @@ type ifReq struct {
 	pad   [0x28 - 0x10 - 2]byte
 }
 
-func ioctl(fd uintptr, request int, argp uintptr) error {
+func ioctl(fd uintptr, request uint, argp uintptr) error {
```

Comment:
Looks like `uintptr` is a better choice, given that `Syscall` expects that exact argument. Would you mind changing to that?

I think I used `int` here originally since this parameter being a pointer felt weird to me, and I just assumed it was always typed as int.